tests/unit/utils/test_gitfs.py

```diff
@@ -89,3 +96,107 @@ def _get_mock(verify, provider):
+            'Create master branch', tree, [])
+        repository.create_reference('refs/tags/simple_tag', commit)
+
+        with salt.utils.files.fopen(os.path.join(repository.workdir, filename), 'w') as file:
+            file.write(filecontent)
+
+        blob = repository.create_blob_fromworkdir(filename)
+        builder = repository.TreeBuilder()
+        builder.insert(
+            filename, blob, os.stat(os.path.join(path, filename)).st_mode)
+        tree = builder.write()
+
+        repository.index.read()
+        repository.index.add(filename)
+        repository.index.write()
+
+        commit = repository.create_commit(
+            'HEAD', signature, signature,
+            'Added a README', tree, [repository.head.target])
+        repository.create_tag(
+            'annotated_tag', commit,
+            pygit2.GIT_OBJ_COMMIT, signature, 'some message')
+
+    def _prepare_cache_repository(self, remote, cache):
+        opts = {
+            'cachedir': cache,
+            '__role': 'minion',
+            'gitfs_disable_saltenv_mapping': False, 'gitfs_base': 'master',
+            'gitfs_insecure_auth': False,
+            'gitfs_mountpoint': '',
+            'gitfs_passphrase': '',
+            'gitfs_password': '',
+            'gitfs_privkey': '',
+            'gitfs_provider': 'pygit2',
+            'gitfs_pubkey': '',
+            'gitfs_ref_types': ['branch', 'tag', 'sha'],
+            'gitfs_refspecs': ['+refs/heads/*:refs/remotes/origin/*', '+refs/tags/*:refs/tags/*'],
+            'gitfs_root': '',
+            'gitfs_saltenv_blacklist': [],
+            'gitfs_saltenv_whitelist': [],
+            'gitfs_ssl_verify': True,
+            'gitfs_update_interval': 3,
+            'gitfs_user': '',
+            'verified_gitfs_provider': 'pygit2'
+        }
+        per_remote_defaults = {
+            'base': 'master',
+            'disable_saltenv_mapping': False,
+            'insecure_auth': False,
+            'ref_types': ['branch', 'tag', 'sha'],
+            'passphrase': '', 'mountpoint': '',
+            'password': '',
+            'privkey': '',
+            'pubkey': '',
+            'refspecs': ['+refs/heads/*:refs/remotes/origin/*', '+refs/tags/*:refs/tags/*'],
+            'root': '',
+            'saltenv_blacklist': [],
+            'saltenv_whitelist': [],
+            'ssl_verify': True,
+            'update_interval': 60,
+            'user': ''
+        }
+        per_remote_only = ('all_saltenvs', 'name', 'saltenv')
+        override_params = tuple(per_remote_defaults.keys())
+        cache_root = cache + '/gitfs'
+        role = 'gitfs'
+        shutil.rmtree(cache_root, ignore_errors=False)
+        provider = salt.utils.gitfs.Pygit2(
+            opts, remote, per_remote_defaults,
+            per_remote_only, override_params, cache_root, role)
+        return provider
+
+    def test_checkout(self):
+        self._prepare_remote_repository('/tmp/pygit2-repo')
```

Comment:
Could you please use `tests.support.paths.TMP` as a base dir for the temporary repository here?

Reply:
Done